### PR TITLE
Send only CONFIG and depending config types when securityconfig is requested

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__

 4. __Why these changes are required?__ 

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from master branch?__ (_If yes, please add backport PR # and commits #_)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/security/action/configupdate/TransportConfigUpdateAction.java [hunk 1]
```diff
@@ -113,7 +116,15 @@ protected ConfigUpdateResponse newResponse(ConfigUpdateRequest request, List<Con
 	
     @Override
     protected ConfigUpdateNodeResponse nodeOperation(final NodeConfigUpdateRequest request) {
-        configurationRepository.reloadConfiguration(CType.fromStringValues((request.request.getConfigTypes())));
+        CType[] allCTypes = CType.values();
+        Set<CType> requestedCTypes = CType.fromStringValues((request.request.getConfigTypes()));
+        Set<CType> requiredCTypes = new HashSet<>();
+        for (CType ct : requestedCTypes) {
+            if (!Arrays.asList(allCTypes).contains(ct))
```
Comment (vrozov) [inline src/main/java/com/amazon/opendistroforelasticsearch/security/action/configupdate/TransportConfigUpdateAction.java:123]: Under what condition `allCTypes` does not contain `ct`?

**Review by vrozov — CHANGES_REQUESTED** [review]
-1: `nodeOperation()` already reloads `CType` included into the `ConfigUpdateRequest`.

**Review by vrozov — CHANGES_REQUESTED** [review]
The solution should cover not only security config but all other CTypes as well.

**Comment by cliu123** [comment]
> The solution should cover not only security config but all other CTypes as well.

I understand that. The follow-up changes will be made before the PR gets published for review.

**Pull request closed without merging**